index.Rmd

```diff
@@ -701,142 +698,63 @@ Refer to the reference section of this document for further information on funct
 
+```{r}
+# Create data and output folders and download data from URL
+data_URL <- "https://github.com/GeoScripting-WUR/Scripting4GeoIntro/releases/download/gadm-data/gadm41_PHL_2.zip"
+data_folder <- "data"
 
-```{r, message=FALSE}
-library(raster)
-?getData
-?raster::getData
-```
-
-Read the help to find out how we can find the country codes.
+if (!dir.exists(data_folder)) {
+  dir.create(data_folder)
+}
 
-```{block, type="alert alert-success"}
-> **Question 8**: What is the country code of Belgium?
+if (!file.exists('data/data.zip')) {
+  download.file(url = data_URL, destfile = 'data/data.zip', method = 'auto', mode = 'wb')
+  unzip('data/data.zip', exdir = 'data')
+}
 ```
 
-More info about the data sets see [Global Adminstrative Areas database (GADM)](http://www.gadm.org).
+We load the administrative boundaries of the Philippines, using the `sf` library, to which you will be introduced more in-depth in a future tutorial:
 
-```{r, eval=FALSE, message=FALSE, results='hide', echo=FALSE}
-raster::getData("ISO3")
-```
+```{r}
+# Check for the terra package and install if missing
+if(!"sf" %in% rownames(installed.packages())){install.packages("sf")}
+library(sf)
 
-Now we will download the administrative boundaries of the Philippines:
+# Load the data and subset its geometry
+adm <- st_read(paste0(data_folder, "/gadm41_PHL_2.json"), quiet = TRUE)
+adm_geom <- st_geometry(adm)
 
-```{r, eval=FALSE}
-datdir <- 'data'
-dir.create(datdir, showWarnings = FALSE)
-adm <- raster::getData("GADM", country = "PHL",
-                       level = 2, path = datdir)
-plot(adm[adm$NAME_1 == "Tarlac",])
-```
-
-```{r loadrda, echo=FALSE, eval=TRUE, message=FALSE}
-datdir <- 'data'
-if (!file.exists(datdir)) { dir.create(datdir)}
-if (!file.exists(file.path(datdir, "PHL_adm2.RData"))) {
-  adm <- raster::getData("GADM", country = "PHL", level=2, path = "data/")
-} else {
-  load(file.path(datdir, "PHL_adm2.RData"))
-  adm <- gadm
-}
+# Create and example plot
+plot(adm_geom[adm$NAME_1 == "Tarlac"])
 ```
 
-Try to understand the code below, and let us know if you have questions. Feel free to use this code as an example and use this for the exercise below.
+Try to understand the code below, and let us know if you have questions. Feel free to use this code as an example and use this for exercise 3.
 
 ```{r phil, eval=TRUE, warning=FALSE, fig.align='center'}
-mar <- adm[adm$NAME_1 == "Marinduque",]
-plot(mar, bg = "dodgerblue", axes=TRUE)
-plot(mar, lwd = 10, border = "skyblue", add=TRUE)
+mar <- adm_geom[adm$NAME_1 == "Marinduque"]
+plot(mar, bg = "dodgerblue", axes = TRUE)
+plot(mar, lwd = 10, border = "skyblue", add = TRUE)
 plot(mar, col = "green4", add = TRUE)
 grid()
 box()
-invisible(text(sp::coordinates(mar),
-labels = as.character(mar$NAME_2), cex = 1.1, col = "white", font = 2))
+invisible(text(st_coordinates(st_centroid(mar)),
+               labels = as.character(adm$NAME_2[adm$NAME_1 == "Marinduque"]), cex = 1.1, col = "white", font = 2))
```

Comment:
Why use `adm` instead of `mar` here?

Reply:
Well, you can keep the whole data.frame rather than just the geometry, and use `st_geometry()` whenever it's necessary.